### PR TITLE
Simplify type annotations

### PR DESCRIPTION
**Pull request by Alnusjaponica**
## Motivation
Resolve https://github.com/optuna/optuna/issues/4508

## Description of the changes
Apply following changes to 
- `optuna_integration/allennlp/`
- `optuna_integration/chainermn.py`
- `optuna_integration/keras.py`
- `tests/allennlp_tests/`
- `tests/test_chainer.py`
- `tests/test_chainermn.py`
- `tests/test_keras.py`

> - `UP007`: Fixes `Union[x, y]` to x | y
> - `UP006`: Fixes collection types (e.g. `List[str]` -> `list[str]`)
> - `F401`: Removes unused imports such as `from typing import List`
> - `I002`: Enforces `from __future__ import annotations`

### CODE REVIEW

File: optuna_integration/chainermn.py [hunk 1]
```diff
@@ -1,11 +1,10 @@
...
+
 from datetime import datetime
 from typing import Any
 from typing import Callable
-from typing import Dict
-from typing import Optional
 from typing import overload
 from typing import Sequence
-from typing import Tuple
 from typing import Type
```
Comment (gen740) [inline optuna_integration/chainermn.py:8]: You can alse remove this line.

Comment (Alnusjaponica) [reply optuna_integration/chainermn.py]: Thank you for your comment. I updated accordingly.

Comment (Alnusjaponica) [reply optuna_integration/chainermn.py]: Thank you for your comments. I updated accordingly.

**Review by gen740 — APPROVED** [review]
LGTM!

**Pull request merged**